airbyte-integrations/connectors/destination-clickhouse-v2/src/main/kotlin/io/airbyte/integrations/destination/clickhouse_v2/write/load/BinaryRowInsertBuffer.kt

```diff
@@ -63,28 +107,77 @@ class BinaryRowInsertBuffer(
                 .insert(
                     "`${tableName.namespace}`.`${tableName.name}`",
                     inner.toInputStream(),
-                    ClickHouseFormat.RowBinary
+                    ClickHouseFormat.RowBinary,
                 )
                 .await()
 
         log.info { "Finished insert of ${insertResult.writtenRows} rows into ${tableName.name}" }
     }
 
-    private fun writeAirbyteValue(columnName: String, abValue: AirbyteValue) {
+    internal fun writeAirbyteValue(
+        columnName: String,
+        abValue: AirbyteValue,
+        dataType: AirbyteType,
+    ) {
+        println(
+            "In the column $columnName pushing value: $abValue with type: ${dataType.javaClass}"
+        )
         when (abValue) {
```

Comment:
nonblocking: are you intentionally not using the EnrichedValue stuff? that class is weird and slow, but it would simplify a lot of what you're doing here

in particular you would need the matchType stuff, b/c EnrichedValue enforces that the AirbyteValue matches the AirbyteType. So you would be able to unconditionally serialize anything that gets written as JSON, and unconditionally not serialize anything that _isn't_ written as JSON

Reply:
This is what Ryan was looking at before he went in PTO.